src/sentry/tasks/statistical_detectors.py

```diff
@@ -53,15 +74,20 @@ def run_detection() -> None:
     performance_projects = []
     profiling_projects = []
 
+    # TODO: make the amount of projects per batch configurable
     for project in RangeQuerySetWrapper(
         Project.objects.filter(status=ObjectStatus.ACTIVE),
         step=100,
     ):
         if project.flags.has_transactions and project.id in enabled_performance_projects:
-            performance_projects.append(project.id)
+            performance_projects.append(project)
 
             if len(performance_projects) >= PROJECTS_PER_BATCH:
-                detect_transaction_trends.delay(performance_projects, now)
+                detect_transaction_trends.delay(
+                    [p.organization_id for p in performance_projects],
+                    [p.id for p in performance_projects],
+                    now,
+                )
```

Comment:
Since we're already querying the projects here, I did not see a reason to query the projects table a second time in another task for the org ids. I can make the same interface change to functions to keep it uniform if people don't have an issue with it

Reply:
The functions query uses the query builder interface which requires the `project` object. So I don't think there's a reason to switch its interface as we have to query the projects table for that.

For the metrics query, if you wish to bypass the query builder and build the query directly like you have below. I don't see any problems with passing the org ids like you have here.

Reply:
got it